src/server/service/kv.rs

```diff
@@ -112,11 +117,31 @@ impl<T: RaftStoreRouter + 'static, E: Engine, L: LockManager> Service<T, E, L> {
         let status = RpcStatus::new(code, Some(format!("{}", err)));
         ctx.spawn(sink.fail(status).map_err(|_| ()));
     }
+
+    fn check_common_names(&self, ctx: &RpcContext) -> bool {
+        if self.x509_common_names.is_empty() {
+            return true;
+        } else if let Some(auth_context) = ctx.auth_context() {
+            if let Some(auth_property) = auth_context
+                .into_iter()
+                .find(|x| x.name() == "x509_common_name")
+            {
+                let peer_cn = auth_property.value_str().unwrap();
+                if self.x509_common_names.contains(peer_cn) {
```

Comment:
Do we need to support wildcard matching?

Reply:
FYI https://tools.ietf.org/html/rfc6125#section-6.4.3

Reply:
Supported